src/components/AgmExtension/ExtensionRequest.vue

```diff
@@ -281,48 +281,117 @@ export default class ExtensionRequest extends Vue {
-          this.data.isEligible = false
-        } else {
-          this.data.isEligible = true
-          const totalExtensionApproved = DateUtilities.subtractDates(
-            DateUtilities.dateToYyyyMmDd(this.data.incorporationDate), this.extensionExpiryDateText) - 18
-          this.data.extensionDuration = Math.min(6, (12 - totalExtensionApproved))
-        }
+      }
+      // ELSE IF CurrentDate > (ExpirationDate + 5 days)
+      // THEN NOT ELIGIBLE
+      else if ((DateUtilities.daysBetweenTwoDates(expiryDate, currentDate)) > 5) {
+        this.data.isEligible = false
+      }
+      // ELSE ELIGIBLE, ExtensionDuration = MIN(6, 12 - totalExtensionApproved)
+      else {
+        this.data.isEligible = true
+        // totalExtensionApproved: (ExpirationDate - 18 months - IncorporationDate) in months
+        const totalExtensionApproved = DateUtilities.subtractDates(
+          DateUtilities.dateToYyyyMmDd(this.data.incorporationDate), this.extensionExpiryDateText) - 18
+        this.data.extensionDuration = Math.min(6, (12 - totalExtensionApproved))
+      }
+    }
+
+    // ELSE IF NOT first agm extension
+    // AND NOT extension requested this year
+    else if (
+      !this.isFirstAgm &&
+      !this.data.isPrevExtension &&
+      this.previousAgmDateText
+    ) {
+      const cutOffYyyyMmDd = DateUtilities.addMonthsToDate(
+        15, this.previousAgmDateText)
+      const cutOffDate = DateUtilities.yyyyMmDdToDate(cutOffYyyyMmDd)
+
+      // IF CurrentDate > (PrevAgmDate + 15 Months + 5 days)
+      // THEN NOT ELIGIBLE
+      if ((DateUtilities.daysBetweenTwoDates(cutOffDate, currentDate) - 5) > 0) {
+        this.data.isEligible = false
+      }
+      // ELSE ELIGIBLE, ExtensionDuration = 6 months
+      else {
+        this.data.isEligible = true
+        this.data.extensionDuration = 6
+      }
+    }
+
+    // ELSE IF NOT first agm extension
+    // AND extension requested this year
+    else if (
+      !this.isFirstAgm &&
+      this.data.isPrevExtension &&
+      this.previousAgmDateText &&
+      this.extensionExpiryDateText
+    ) {
+      const cutOffDate = DateUtilities.yyyyMmDdToDate(this.previousAgmDateText)
+      const expiryDate = DateUtilities.yyyyMmDdToDate(this.extensionExpiryDateText)
+
+      // IF CurrentDate > (ExpirationDate + 5 days)
+      // THEN NOT ELIGIBLE
+      if ((DateUtilities.daysBetweenTwoDates(expiryDate, currentDate)) > 5) {
+        this.data.isEligible = false
+      }
+      // ELSE IF (ExpirationDate - PrevAgmDate) > 12 months
+      // THEN NOT ELIGIBLE
+      else if ((DateUtilities.daysBetweenTwoDates(cutOffDate, expiryDate)) > 365) {
+        this.data.isEligible = false
+      }
+      // ELSE ELIGIBLE, ExtensionDuration = 12 months - ExpiryDate - PrevAgmDate
+      else {
+        this.data.isEligible = true
+        this.data.extensionDuration = 12 - DateUtilities.subtractDates(
+          this.previousAgmDateText,
+          this.extensionExpiryDateText
+        )
```

Comment:
In the next PR, let's see about moving some of these blocks into other functions, with the goal of keeping the "high level" logic in here and moving the "details" to the other functions.

Reply:
I agree, even with the comments it is still difficult to understand what is going on at a glance. Moving some of the functionality out of the blocks to other functions should really help this.